modules/node-shared/src/main/scala/org/tessellation/node/shared/domain/cluster/programs/Joining.scala

```diff
@@ -238,7 +205,7 @@ sealed abstract class Joining[F[_]: Async: GenUUID: SecurityProvider: Hasher] pr
           localHealthcheck.cancel(registrationRequest.id),
           PeerAlreadyJoinedWithDifferentRegistrationData(registrationRequest.id).raiseError[F, Unit]
         )
-      _ <- nodeStorage.tryModifyStateGetResult(NodeState.SessionStarted, stateAfterJoining)
+      _ <- nodeStorage.tryModifyStateGetResult(SessionStarted, stateAfterJoining)
```

Comment:
I think the whole point of this functionality was to postpone this state change until we attempt to join to all discovered peers. With this line of code the switch will still happen after joining to a first node so exactly as it was before. I think you need to remove it.

Reply:
@TheMMaciek , you are correct. This line has been removed.